course3/homework-list/list.h

```diff
@@ -0,0 +1,105 @@
+#pragma once
+
+#include <memory>
+
+
+namespace Karun
+{
+
+template<typename T>
+class List
+{
+public:
+	class Node
+	{
+	friend class List;
+	
+	public:
+		Node(const T& data, std::unique_ptr<Node>& next)
+		: 	mData(data)
+		{
+			mNext = std::move(next);
+		}
+		
+	private:
+		T mData;
+		std::unique_ptr<Node> mNext;
+	};
+	
+	class iterator
+	{
+	friend class List;
+		
+	public:
+		iterator(Node* node) : mNode(node) {}
```

Comment:
Is it okay to pass raw pointers around for iterators, or would smart pointers be better? If the latter, would the correct solution be to replace unique_ptrs with shared_ptrs, and use weak_ptrs for iterators? I thought about doing this, but didn't implement it since it seemed like overkill.